### ISSUE TITLE
Datepicker - disable validation option

### ISSUE THREAD

**Issue by Blackbaud-PatrickOFriel** [angular-ui/bootstrap#3836]
I have a situation where in some cases where I am using the datepicker popup and I don't want the manually entered input to go through the datepicker's validation (I want to do custom validation instead). Are there any plans to have either an option to disable validation on the datepicker or an option to provide custom validation for the datepicker (or maybe there's already a way that I'm too thick to discover today)? Thanks for all your work!

**Comment by adouglas** [comment]
+1

**Comment by wesleycho** [comment]
I am currently in the process of working on a fix to the validation of the datepicker component [here](https://github.com/wesleycho/bootstrap/tree/fix/datepicker-validators) - once this is implemented, one can write a custom directive with a `$timeout` hack to overwrite `ngModelCtrl.$validators`, allowing for custom validators.

**Comment by Blackbaud-PatrickOFriel** [comment]
I found a way to hack around the current validation (albeit in a very strange way), my main pain point at the moment is this guy popping up angry and red in the console when I'm doing my own validation: `$log.error('Datepicker directive: "ng-model" value must be a Date object, a number of milliseconds since 01.01.1970 or a string representing an RFC2822 or ISO 8601 date.');`. Any way this can be a warning instead of an error?

**Comment by Blackbaud-PatrickOFriel** [comment]
Any more info on whether we can have that $log.error be a warning, or better yet, not exist? I've managed to hack my own validation, but that console error keeps coming up.

**Comment by wesleycho** [comment]
One can disable `$log` via

```javascript
$provide.factory('$log', function() {
  return {
    debug: angular.noop,
    error: angular.noop,
    log: angular.noop,
    warn: angular.noop
  };
});

**Comment by Blackbaud-PatrickOFriel** [comment]
I thought about doing that, but I really only wanted to disable the validation for this case, since I know that I'm overriding the typical validation here. I dont want to suppress all log messages in my application. I'll try to think of another way to work around it.

**Comment by wesleycho** [comment]
One way is to file a PR that creates an injectable `$datepickerSuppressError` and `$dateparserSuppressWarning` and the $log usage in those components can then check for the truth condition. similar to how it is done in `$transition`.

**Comment by Blackbaud-PatrickOFriel** [comment]
Okay I'll take a look at that, thanks!

**Comment by distinctdan** [comment]
I too, am in need of disabling the validation. I set the entered value to the api on the model, even if the user has entered stuff that doesn't exactly match the format, like leading 0's for the day and month. Could we add a disable-validation attribute that turns off the datepicker's validation to allow setting invalid values?

Note that I've tried `ng-model-options="{ allowInvalid: true }"`, but it doesn't have any effect.
